purchase_consignment_inventory/models/stock_inventory.py

```diff
@@ -29,39 +19,57 @@ class StockInventory(models.Model):
 
     @api.onchange('filter')
     def _onchange_filter__if_consignment__set_exhausted(self):
-        if self.filter == CONSIGNMENT_FILTER_KEY:
+        if self.filter == CONSIGNMENT:
             self.exhausted = True
 
     @api.onchange('filter')
     def _onchange_filter__if_not_consignment__unselect_supplier(self):
-        if self.filter != CONSIGNMENT_FILTER_KEY:
+        if self.filter != CONSIGNMENT:
             self.consignment_supplier_id = None
 
     @api.model
     def _selection_filter(self):
         selection = super()._selection_filter()
-        selection.append((CONSIGNMENT_FILTER_KEY, _('Consignment Supplier')))
+        selection.append((CONSIGNMENT, _('Consignment Supplier')))
         return selection
 
+    def _get_inventory_lines_values(self):
+        if self.filter == CONSIGNMENT:
+            return self._get_consignment_inventory_lines_values()
+        else:
+            return super()._get_inventory_lines_values()
+
+    def _get_consignment_inventory_lines_values(self):
+        all_products = self._get_all_products_for_consignment_supplier()
+        quants = self._get_all_quants_in_inventoried_location(all_products)
+        result = self._get_line_values_from_quants(quants)
+
+        if self.exhausted:
+            products_not_exhausted = quants.mapped('product_id')
+            exhausted_products = all_products - products_not_exhausted
+            result.extend(self._get_exhausted_consignment_line_values(exhausted_products))
+
+        return result
+
     def _get_all_products_for_consignment_supplier(self):
         supplier_info = self.env['product.supplierinfo'].search([
             ('name.commercial_partner_id', '=', self.consignment_supplier_id.id),
         ])
         all_products = get_products_from_supplier_info(supplier_info)
         return all_products.filtered(lambda p: p.consignment)
 
-    def _get_consignment_inventory_lines_values(self):
+    def _get_all_quants_in_inventoried_location(self, products):
         locations = self.env['stock.location'].search([('id', 'child_of', [self.location_id.id])])
-        all_products = self._get_all_products_for_consignment_supplier()
-        all_quants = self.env['stock.quant'].search([
-            ('product_id', 'in', all_products.ids),
+        return self.env['stock.quant'].search([
+            ('product_id', 'in', products.ids),
             ('location_id', 'in', locations.ids),
         ])
 
+    def _get_line_values_from_quants(self, quants):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-purchase-addons/pullRequest?prid=4908134)